policy-service/src/component-tests/java/uk/gov/gchq/palisade/component/policy/model/AuditablePolicyRecordResponseTest.java

```diff
@@ -0,0 +1,138 @@
+    void testAuditablePolicyRecordResponseSerialisingAndDeserialising() throws IOException {
+        AuditablePolicyRecordResponse auditablePolicyRecordResponse = AuditablePolicyRecordResponse.Builder.create().withPolicyResponse(RESPONSE).withNoErrors();
+
+        JsonContent<AuditablePolicyRecordResponse> auditablePolicyRecordResponseJsonContent = jsonTester.write(auditablePolicyRecordResponse);
+        ObjectContent<AuditablePolicyRecordResponse> auditablePolicyRecordResponseObjectContent = jsonTester.parse(auditablePolicyRecordResponseJsonContent.getJson());
+        AuditablePolicyRecordResponse auditablePolicyRecordResponseMessageObject = auditablePolicyRecordResponseObjectContent.getObject();
+        assertAll("AuditSerialisingDeseralisingAndComparison",
+                () -> assertAll("AuditSerialisingComparedToString",
+                        () -> assertThat(auditablePolicyRecordResponseJsonContent).extractingJsonPathStringValue("$.policyResponse.userId").isEqualTo("test-user-id"),
+                        () -> assertThat(auditablePolicyRecordResponseJsonContent).extractingJsonPathStringValue("$.auditErrorMessage").isNull()
+                ),
+
+                () -> assertAll("AuditDeserialisingComparedToObject",
+                        () -> assertThat(auditablePolicyRecordResponseMessageObject.getPolicyResponse()).isEqualTo(auditablePolicyRecordResponse.getPolicyResponse()),
+                        () -> assertThat(auditablePolicyRecordResponseMessageObject.getAuditErrorMessage()).isNull(),
+                        () -> assertThat(auditablePolicyRecordResponse.getAuditErrorMessage()).isNull()
+
+                ),
+                () -> assertAll("ObjectComparison",
+                        //The reconstructed stack trace wont be exactly the same due to different object hashes so equals is used here
+                        () -> assertThat(auditablePolicyRecordResponseMessageObject).usingRecursiveComparison().isEqualTo(auditablePolicyRecordResponse)
+                )
+        );
+    }
+
+    /**
+     * Grouped assertion test for a {@link AuditablePolicyRecordResponse} which holds an exception an no policy response
+     * This is the scenario where the object represents a error that is to be passed onto the audit service.
+     * Create the object with the builder and then convert to the Json equivalent.
+     * Takes the JSON Object, deserialises and tests against the original Object
+     *
+     * @throws IOException throws if the {@link AuditablePolicyRecordResponse} object cannot be converted to a JsonContent.
+     *                     This equates to a failure to serialise or deserialise the string.
+     */
+    @Test
+    void testAuditablePolicyRecordResponseExceptionSerialisingAndDeserialising() throws IOException {
+        AuditablePolicyRecordResponse auditablePolicyRecordResponse = AuditablePolicyRecordResponse.Builder.create().withPolicyResponse(null).withAuditErrorMessage(AUDIT_ERROR_MESSAGE);
```

Comment:
I have added  the missing equals() and hashCode() to the class.  I have included both the recursive comparison and the isEqualTo to each of the tests. I have started to add the self documenting  .as(text) to the methods in the test class will adopt that going forward.

I like the minimalism that you have suggested for this test, but I am going to hold off on deleting the existing assertions for the moment.  I was asked to put in the finer detailed assertions for each attribute in a code review that was previously on the classes in the model package and this became convention.  We want to change this, I feel that we need to run it up the flag pole.

Reply:
Yeah, this bit of code should be discussed. I think the reason for doing it is so that if the equality check failes, the test highlights which field. But by using `.usingRecursiveComparison()` and then doing an equals check, AssertJ will indicate in the test which fields did not match. The good thing here is that if another field is added, it will automatically be included in the test :)